### PR TITLE
Hook Docs for lib/endpoints/class-wp-rest-users-controller.php

### PR DESCRIPTION
**Pull request by adamsilverstein**
See https://github.com/WP-API/WP-API/issues/1549.

* Add some missing hook docs
* Spacing, punctuation cleanup as per standard
* Change second do_action from `rest_insert_user` to `rest_update_user` as
its in the update function
* added some todo notes, I don’t understand what the final false
variable is for, maybe its needed?

### CODE REVIEW

File: lib/endpoints/class-wp-rest-users-controller.php [hunk 1]
```diff
@@ -235,11 +236,12 @@ public function create_item( $request ) {
...
+		 * Fires after a user is created via the REST API.
 		 *
-		 * @param object $user Data used to create user (not a WP_User object)
+		 * @param object          $user    Data used to create the user (not a WP_User object).
 		 * @param WP_REST_Request $request Request object.
-		 * @param bool $bool A boolean that is false.
+		 * @param bool            $bool    A boolean that is false.
+		 * @todo remove final bool = false variable, unused?
 		 */
 		do_action( 'rest_insert_user', $user, $request, false );
```
Comment (rachelbaker) [inline lib/endpoints/class-wp-rest-users-controller.php]: @adamsilverstein The third param here should be `true` on "Create" (it should be false on update).

File: lib/endpoints/class-wp-rest-users-controller.php [hunk 2]
```diff
@@ -299,7 +301,15 @@ public function update_item( $request ) {
...
-		do_action( 'rest_insert_user', $user, $request, false );
+		/**
+		 * Fires after a user is updated via the REST API.
+		 *
+		 * @param object          $user    Data used to update the user (not a WP_User object).
+		 * @param WP_REST_Request $request Request object.
+		 * @param bool            $bool    A boolean that is false.
+		 * @todo remove final bool = false variable, unused?
+		 */
+		do_action( 'rest_update_user', $user, $request, false );
```
Comment (rachelbaker) [inline lib/endpoints/class-wp-rest-users-controller.php]: @adamsilverstein to keep the same naming convention in the other endpoint controllers, it should be `rest_insert_user` but with the third param of `false`.

File: lib/endpoints/class-wp-rest-users-controller.php (hunk 1 above)
Comment (adamsilverstein) [inline lib/endpoints/class-wp-rest-users-controller.php]: Thanks for clarifying, I will add that info to the docblock

File: lib/endpoints/class-wp-rest-users-controller.php (hunk 2 above)
Comment (adamsilverstein) [inline lib/endpoints/class-wp-rest-users-controller.php]: I'm learning so much already :) Thanks, will fix the docblock.

**Comment by adamsilverstein** [comment]
Note: I believe this also corrects an issue with the first call of the `rest_insert_user` filter inside `create_item`, I changed the last ("created") parameter to true as per @rachelbaker's comment on my previous commit (https://github.com/WP-API/WP-API/pull/1556#discussion_r39616194). 

Please confirm this is correct before merging :)

**Comment by adamsilverstein** [comment]
Working on lib/endpoints/class-wp-rest-terms-controller.php now and wondering why I don't see similar do_action( 'rest_insert_term' ...) hooks firing. Is this intentional, or an oversight? If the latter I can go ahead and add them as I go :)

**Comment by rachelbaker** [comment]
> Working on lib/endpoints/class-wp-rest-terms-controller.php now and wondering why I don't see similar do_action( 'rest_insert_term' ...) hooks firing. Is this intentional, or an oversight? If the latter I can go ahead and add them as I go :)

@adamsilverstein Just an oversight.

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,5 @@
 composer.lock
 node_modules
 vendor
+
+.idea/
```
Comment (rachelbaker) [inline .gitignore]: @adamsilverstein can you undo this change to the .gitignore?

Comment (adamsilverstein) [inline .gitignore]: Ugh, should be in my local exclude sorry. kind of switching back to sublime anyway :)

Comment (adamsilverstein) [inline .gitignore]: Removed.

**Comment by adamsilverstein** [comment]
@rachelbaker This should be good to go unless I missed some feedback, I'm addressing adding missing hooks in https://github.com/WP-API/WP-API/issues/1571

**Comment by rmccue** [comment]
:sparkles:
